src/panel/panel.vala

```diff
@@ -533,15 +541,38 @@ namespace Budgie {
 								expected_uuids.remove_link(g);
 							}
 							message("Unable to load invalid applet: %s", applets[i]);
+							applet_removed(applets[i]);
+							continue;
+						}
+
+						info = this.add_pending(applets[i], name);
+						if (info == null) {
 							continue;
 						}
-						this.add_pending(applets[i], name);
-						manager.modprobe(name);
+					}
+
+					/* um add this bro to the panel :o */
+					if (info.alignment == "start") {
+						start_applets.insert_sorted(info, infocmp);
+					} else if (info.alignment == "center") {
+						center_applets.insert_sorted(info, infocmp);
 					} else {
-						/* um add this bro to the panel :o */
-						this.add_applet(info);
+						end_applets.insert_sorted(info, infocmp);
 					}
 				}
+
+				for (int i = 0; i < start_applets.length(); i++) {
+					start_applets.nth_data(i).position = i;
+					add_applet(start_applets.nth_data(i));
+				}
+				for (int i = 0; i < center_applets.length(); i++) {
+					center_applets.nth_data(i).position = i;
+					add_applet(center_applets.nth_data(i));
+				}
+				for (int i = 0; i < end_applets.length(); i++) {
+					end_applets.nth_data(i).position = i;
+					add_applet(end_applets.nth_data(i));
+				}
```

Comment:
Worth maybe doing some g_list_free at the end on the start, center; and end applets?

https://valadoc.org/glib-2.0/GLib.g_list_free.html

Reply:
The lists are freed once the function finishes, so this shouldn't be necessary.